Update to go1.22 (#10415)
traefik/traefik: docs/content/migration/v2.md

### IPWhiteList (TCP)

In `v2.11`, the `IPWhiteList` middleware is deprecated, please use the [IPAllowList](../middlewares/tcp/ipallowlist.md) middleware instead.

### TLS CipherSuites

> By default, cipher suites without ECDHE support are no longer offered by either clients or servers during pre-TLS 1.3 handshakes.
> This change can be reverted with the `tlsrsakex=1 GODEBUG` setting.

> > https://go.dev/doc/go1.22#crypto/tls
> 
> The following ciphers has been removed from the default: `TLS_RSA_WITH_AES_128_CBC_SHA`, `TLS_RSA_WITH_AES_256_CBC_SHA`, `TLS_RSA_WITH_AES_128_GCM_SHA256`, `TLS_RSA_WITH_AES_256_GCM_SHA384`.
> 
> If you need them you have to either define them into your TLS configuration or set the environment variable `GODEBUG=tlsrsakex=1`.  

Reviewer: ```suggestion
> (https://go.dev/doc/go1.22#crypto/tls)

The following ciphers have been removed from the default list:

- `TLS_RSA_WITH_AES_128_CBC_SHA`
- `TLS_RSA_WITH_AES_256_CBC_SHA`
- `TLS_RSA_WITH_AES_128_GCM_SHA256`
- `TLS_RSA_WITH_AES_256_GCM_SHA384`

To enable these ciphers, please set the option `CipherSuites` in your [TLS configuration](https://doc.traefik.io/traefik/https/tls/#cipher-suites) or set the environment variable `GODEBUG=tlsrsakex=1`.
```

### Minimum TLS Version

> By default, the minimum version offered by `crypto/tls` servers is now TLS 1.2 if not specified with config.MinimumVersion,
> matching the behavior of crypto/tls clients.

> > This change can be reverted with the `tls10server=1 GODEBUG` setting
> > https://go.dev/doc/go1.22#crypto/tls
> 
> If you need to use TLS 1.0, you should either define explicitly your `MinVersion` to `VersionTLS10` or set the environment variable `GODEBUG=tls10server=1`.

Reviewer: ```suggestion
> This change can be reverted with the `tls10server=1 GODEBUG` setting.
> (https://go.dev/doc/go1.22#crypto/tls)

To enable TLS 1.0, please set the option `MinVersion` to `VersionTLS10` in your [TLS configuration](https://doc.traefik.io/traefik/https/tls/#cipher-suites) or set the environment variable `GODEBUG=tls10server=1`.
```

### IPWhiteList (TCP)

In `v2.11`, the `IPWhiteList` middleware is deprecated, please use the [IPAllowList](../middlewares/tcp/ipallowlist.md) middleware instead.

### TLS CipherSuites

> > By default, cipher suites without ECDHE support are no longer offered by either clients or servers during pre-TLS 1.3 handshakes.

Reviewer 2: We should give more explanations on why we do this.

Author: This is a quote from the Go release.

Maybe you suggest improving the sentence bellow this quote?

> > By default, cipher suites without ECDHE support are no longer offered by either clients or servers during pre-TLS 1.3 handshakes.

Reviewer 2: ```suggestion
> By default, cipher suites without ECDHE support are no longer offered by either clients or servers during pre-TLS 1.3 handshakes. The _RSA key exchange_ cipher suites are way less secure than the modern ECDHE cipher suites and exposes potential vulnerabilities like [the Marvin Attack](https://people.redhat.com/~hkario/marvin). Decision has been made to support ECDHE cipher suites only by default.
```
